ext/scheduler/airflow2/client.go

```diff
@@ -0,0 +1,166 @@
+package airflow2
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/odpf/optimus/models"
+)
+
+const (
+	Hundred = 100
+)
+
+type airflowRequest struct {
+	URL    string
+	method string
+	body   []byte
+	param  string
+}
+
+type DagRunList struct {
+	DagRuns      []DagRun `json:"dag_runs"`
+	TotalEntries int      `json:"total_entries"`
+}
+
+type DagRun struct {
+	DagRunID        string    `json:"dag_run_id"`
+	DagID           string    `json:"dag_id"`
+	LogicalDate     time.Time `json:"logical_date"`
+	ExecutionDate   time.Time `json:"execution_date"`
+	StartDate       time.Time `json:"start_date"`
+	EndDate         time.Time `json:"end_date"`
+	State           string    `json:"state"`
+	ExternalTrigger bool      `json:"external_trigger"`
+	Conf            struct{}  `json:"conf"`
+}
+
+type DagRunReqBody struct {
+	OrderBy          string   `json:"order_by"`
+	PageOffset       int      `json:"page_offset"`
+	PageLimit        int      `json:"page_limit"`
+	DagIds           []string `json:"dag_ids"`
+	ExecutionDateGte string   `json:"execution_date_gte,omitempty"`
+	ExecutionDateLte string   `json:"execution_date_lte,omitempty"`
+}
+
+type airflowClient struct {
+	client HTTPClient
+}
+
+func newClient(client HTTPClient) airflowClient {
```

Comment:
the newClient method is not necessary, we can expect users to  directly initialize the struct

Reply:
removed in latest commit